fed/mosaic-cell/src/main/java/org/eclipse/mosaic/fed/cell/utility/RegionUtility.java

```diff
@@ -150,45 +146,61 @@ public static CNetworkProperties getRegionForPosition(CartesianPoint position) {
+                    regions.add(region);
+                }
+            }
         } else {
-            destPolygon = ((GeoRectangle) geoArea).toCartesian().toPolygon();
-        }
-
-        List<CNetworkProperties> regions = new ArrayList<>();
-        for (CMobileNetworkProperties region : ConfigurationData.INSTANCE.getRegionConfig().regions) {
-            if (regionContainsOrIntersectsWithDestinationArea(region, destPolygon)) {
-                regions.add(region);
+            if (geoArea instanceof GeoRectangle) {
+                geoArea = ((GeoRectangle) geoArea).toPolygon();
+            }
+            CartesianPolygon destPolygon = ((GeoPolygon) geoArea).toCartesian();
+            for (CMobileNetworkProperties region : ConfigurationData.INSTANCE.getRegionConfig().regions) {
+                if (region.getCapoArea().isIntersectingPolygon(destPolygon)) {
+                    regions.add(region);
+                }
             }
         }
 
         return regions;
     }
 
-    private static boolean regionContainsOrIntersectsWithDestinationArea(CMobileNetworkProperties regionalProperties,
-                                                                         CartesianPolygon destinationArea) {
-        // TODO might be slow, check if there's a better algorithm and implement it in mosaic-geomath
-        if (regionalProperties.getCapoArea() == null) {
-            return false;
-        }
-
-        for (CartesianPoint corner : destinationArea.getVertices()) {
-            if (regionalProperties.getCapoArea().contains(corner)) {
-                return true;
-            }
+    /**
+     * Collision detection for the collision of a circle with a polygon.
+     *
+     * @param destinationArea The circle.
+     * @param regionalArea The polygon.
+     * @return true if the circle and the polygon collide.
+     */
+    private static boolean isCircleIntersectingPolygon(CartesianCircle destinationArea,
+                                                        CartesianPolygon regionalArea) {
+        // Check if arbitrary point of one area is contained within the other
+        if (regionalArea.contains(destinationArea.getCenter()) || destinationArea.contains(regionalArea.getVertices().get(0))) {
+            return true;
         }
-        for (CartesianPoint otherCorner : regionalProperties.getCapoArea().getVertices()) {
-            if (destinationArea.contains(otherCorner)) {
+        // Check if any edge of the regionalArea intersects the circular destinationArea
+        CartesianPoint lastPoint = regionalArea.getVertices().get(regionalArea.getVertices().size()-1);
+        CartesianPoint circleCenter = destinationArea.getCenter();
+        for (CartesianPoint point : regionalArea.getVertices()) {
+            double dx = lastPoint.getX() - point.getX();
+            double dy = lastPoint.getY() - point.getY();
+            double distanceFromCircleCenter =
```

Comment:
Isn't that just the same as `point.distanceTo(circleCenter)`?

Reply:
No, it directly calculates the smallest distance between a vertice and the circle center.

Reply:
ah, you mean "it directly calculates the smallest distance between **an edge** and the circle center."?

Reply:
Yes right, an edge not vertice :)